std::binary_function is deprecated in C++11

When I compile Globulation 2 with g++ 12.1.0 in MSys2 Mingw64 shell, it tells me that std::binary_function is deprecated. [This Stack Overflow answer](https://stackoverflow.com/a/33115140/8890345) says it can be replaced by three using statements. [cppreference](https://en.cppreference.com/w/cpp/utility/functional/binary_function) says it was deprecated in C++11 and removed in C++17.
```
g++ -o build/libgag/src/GUIFileList.o -c -g3 -Wall -fPIC -DHAVE_CONFIG_H -Dmain=SDL_main -D_GNU_SOURCE=1 -Dmain=SDL_main -I/usr/local/include/SDL -I/mingw64/include -I/mingw64/include/SDL2 -Ilibgag/include -I. -Ilibusl/src -Ibuild/libgag/src/D:/msys64/mingw64/include/SDL2 -Ilibgag/src/D:/msys64/mingw64/include/SDL2 libgag/src/GUIFileList.cpp
libgag/src/GUIFileList.cpp:175:49: warning: 'template<class _Arg1, class _Arg2, class _Result> struct std::binary_function' is deprecated [-Wdeprecated-declarations]
  175 |         struct strfilecmp_functor : public std::binary_function<std::string, std::string, bool>
      |                                                 ^~~~~~~~~~~~~~~
In file included from D:/msys64/mingw64/include/c++/12.1.0/bits/refwrap.h:39,
                 from D:/msys64/mingw64/include/c++/12.1.0/vector:66,
                 from libgag/include/FileManager.h:24,
                 from libgag/include/GUIFileList.h:23,
                 from libgag/src/GUIFileList.cpp:20:
D:/msys64/mingw64/include/c++/12.1.0/bits/stl_function.h:131:12: note: declared here
  131 |     struct binary_function
      |            ^~~~~~~~~~~~~~~
```

--- a/libgag/src/GUIFileList.cpp
+++ b/libgag/src/GUIFileList.cpp
@@ -172,9 +172,9 @@ namespace GAGGUI
 		return fullName;
 	}
 	
-	struct strfilecmp_functor : public std::binary_function<std::string, std::string, bool>
+	struct strfilecmp_functor
 	{
-		bool operator()(std::string x, std::string y)
+		bool operator()(const std::string &x, const std::string &y)
 		{
 			bool xIsNotDir = (x[x.length()-1] != DIR_SEPARATOR);
 			bool yIsNotDir = (y[y.length()-1] != DIR_SEPARATOR);

--- a/src/EndGameScreen.cpp
+++ b/src/EndGameScreen.cpp
@@ -320,7 +320,7 @@ void EndGameStat::onSDLMouseMotion(SDL_Event* event)
 
 
 //! This function is used to sort the player array
-struct MoreScore : public std::binary_function<const TeamEntry&, const TeamEntry&, bool>
+struct MoreScore
 {
 	EndOfGameStat::Type type;
 	bool operator()(const TeamEntry& t1, const TeamEntry& t2)

--- a/src/MapHeader.cpp
+++ b/src/MapHeader.cpp
@@ -286,7 +286,7 @@ std::string glob2FilenameToName(const std::string& filename)
 }
 
 template<typename It, typename T>
-class contains: std::unary_function<T, bool>
+class contains
 {
 public:
 	contains(const It from, const It to) : from(from), to(to) {}

